Sources/Utility/Process.swift

```diff
@@ -165,6 +172,24 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Validate that the program provided exists on the search path and is executable.
+    internal func validateArgumentIsExecutable(_ program: String) throws {
+
+        // check to see if the argument is already listed in the vaildated cache
+        guard !Process.validatedExecutables.contains(program) else {
+            return
+        }
+        let envSearchPaths = Utility.getEnvSearchPaths(pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
+        guard let executableLocation = Utility.lookupExecutablePath(filename: program, searchPaths: envSearchPaths) else {
+            throw Process.Error.missingProgram(program: program)
+        }
+        guard localFileSystem.isExecutableFile(executableLocation) else {
```

Comment:
In general, feel free to rename a methods to clarify their intents.

For this specific case, I actually don't want a specific error because there could be a lot of reason for failure... it could be a dir, we might not have permission, the mode of file may be invalid. Its probably better to just say "we couldn't find the executable you wanted" even tho thats a bit vague. What do you think?